lightclient/src/client.rs

```diff
@@ -46,22 +79,59 @@ impl LightClientRpc {
             .add_chain(config)
             .map_err(|err| LightClientRpcError::AddChainError(err.to_string()))?;
 
-        let (to_backend, backend) = mpsc::unbounded_channel();
-
-        // `json_rpc_responses` can only be `None` if we had passed `json_rpc: Disabled`.
         let rpc_responses = json_rpc_responses.expect("Light client RPC configured; qed");
 
+        let raw_client = Self::new_from_client(
+            client,
+            iter::once(AddedChain {
+                chain_id,
+                rpc_responses,
+            }),
+        );
+        Ok(raw_client.for_chain(chain_id))
+    }
+
+    /// Constructs a new [`RawLightClientRpc`] from the raw smoldot client.
+    ///
+    /// Receives a list of RPC objects as a result of calling `smoldot_light::Client::add_chain`.
+    /// This [`RawLightClientRpc`] can target different chains using [`RawLightClientRpc::for_chain`] method.
+    ///
+    /// ## Panics
+    ///
+    /// ### Native
+    ///
+    /// Panics when called outside of `tokio` runtime context for native context.
```

Comment:
I wonder if there's any sensible way to hide tokio, so that users don't need to care, and we can potentially remove it as a dependency in the future. Possibly not though; nothing comes to mind offhand!